cpu/atmega_common/periph/gpio.c

```diff
@@ -254,6 +428,77 @@ static inline void irq_handler(uint8_t int_num)
     __exit_isr();
 }
 
+#ifdef PCINT_NUM_BANKS
+/* inline function that is used by the PCINT ISR */
+static inline void pcint_handler(uint8_t bank, volatile uint8_t *mask_reg)
+{
+    __enter_isr();
+    /* Find right item */
+    uint8_t pin_num = 0;
+    uint8_t enabled_pcints = *mask_reg;
+
+    while (enabled_pcints > 0) {
+        /* check if this pin is enabled & has changed */
+        if (enabled_pcints & 0x1) {
+            /* re-construct mask from pin number*/
+            uint8_t pin_mask = (1 << pin_num);
```

Comment:
This does not work, as `pin_num` is the idx in the PCINT mapping, not the pin number in the GPIO port. (Those happen to be the same for all but the ATmega2560.)

Here is my suggestion to fix this: https://github.com/maribu/RIOT/commit/8f28ebd73272a4dac4b112b6fa2a3ad8c348ddd3

Reply:
done.